File: CalculatorJava/src/main/java/mx/tecnm/culiacan/project/calculator/controller/CalculatorController.java
package mx.tecnm.culiacan.project.calculator.controller;

import mx.tecnm.culiacan.project.calculator.model.CalculatorService;
import mx.tecnm.culiacan.project.calculator.model.Operation;
import mx.tecnm.culiacan.project.calculator.view.CalculatorView;

import java.awt.event.ActionEvent;
import java.awt.event.ActionListener;

public class CalculatorController implements ActionListener {

    private final CalculatorView view;
    private final CalculatorService model;

    public CalculatorController(CalculatorView view, CalculatorService model){
        this.view = view;
        this.model = model;
    }

    @Override
    public void actionPerformed(ActionEvent e) {
        int number = view.getNumber();

        if (e.getSource() == view.getBtnFactorial()) {
            calculate("Factorial",number);
            return;
        }

        if (e.getSource() == view.getBtnFibonacci()) {
            calculate("Fibonacci",number);
            return;
        }

        if (e.getSource() == view.getBtnAckermann()) {
            calculate("Ackermann",number);
            return;
        }

        if (e.getSource() == view.getBtnClean()) {
            view.clean();
        }
    }

    private Operation calculate (String operation, int number){
        Operation op = model.processOperation(operation, number);
        view.setResult(op);
        return op;
    }
}

File: app/Http/Controllers/CalculatorController.php
<?php

namespace App\Http\Controllers;

use Illuminate\Http\Request;        
use App\Services\CalculatorService;

class CalculatorController extends Controller
{

    private $calculatorService;

    public function __construct(CalculatorService $calculatorService)
    {
        $this->calculatorService = $calculatorService;
    }

    public function calculator_home(){
        return view('calculator.calculator_view');
    }

    public function calculator_process(Request $request)
    {
        $option = $request->input('option');
        $number = $request->input('number');

        $operation = null;

        if ($option !== 'limpiar' && $number !== null) {
            $operation = $this->calculatorService->processOperation($option, $number);
        }

        return view('calculator.calculator_view', compact('operation'));
}

}